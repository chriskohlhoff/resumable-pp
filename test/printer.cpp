#include <iostream>
#include <utility>

auto print_1_to(int n)
{
  return lambda_initializer(
    [n] resumable {
      for (int i = 1;;)
      {
        std::cout << i << std::endl;
        if (++i > n) break;
        break_resumable;
      }
    }
  );
}

int main()
{
  auto i = print_1_to(10);
  initializer_lambda<decltype(i)> r(std::move(i));
  while (!ready(r))
  {
    std::cout << "resuming ... ";
    resume(r);
  }
}
