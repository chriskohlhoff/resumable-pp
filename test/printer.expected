resuming ... 1
resuming ... 2
resuming ... 3
resuming ... 4
resuming ... 5
resuming ... 6
resuming ... 7
resuming ... 8
resuming ... 9
resuming ... 10
